### PR TITLE
added sourceBox parameter to resizeInto in RasterYuv420Semiplanar

### PR DESCRIPTION
**Pull request by sebastianbaginski**
This allows to crop & resize in single pass.
@marcusnaslund review

### CODE REVIEW

File: source/draw/RasterYuv420Semiplanar.ooc [hunk 1]
```diff
@@ -92,30 +92,36 @@ RasterYuv420Semiplanar: class extends RasterImage {
 		}
 		result
 	}
-	resizeInto: func (target: This) {
+	resizeInto: func (target: This, sourceBox := IntBox2D new()) {
+		if (sourceBox hasZeroArea)
+			sourceBox = IntBox2D new(this size)
```
Comment (marcusnaslund) [inline source/draw/RasterYuv420Semiplanar.ooc]: I suppose `sourceBox := IntBox2D new(this size)` does not work as a default argument?

File: test/draw/RasterYuv420SemiplanarTest.ooc [hunk 2]
```diff
@@ -122,6 +122,16 @@ RasterYuv420SemiplanarTest: class extends Fixture {
...
+		this add("resize (with box)", func {
+			source := RasterYuv420Semiplanar open(this _inputPath)
+			box := IntBox2D new(0, source height / 2, source width, source height / 2)
+			target := RasterYuv420Semiplanar new(source size)
+			source resizeInto(target, box)
+			output := "test/draw/output/RasterYuv420SemiplanarTest_lowerHalfResized.png"
+			target save(output)
+			(target, source) referenceCount decrease()
+			output free()
+		})
```
Comment (marcusnaslund) [inline test/draw/RasterYuv420SemiplanarTest.ooc]: I see the other tests are just as "bad", but yeah, this test only calls the function, it doesn't check the output at all.

File: source/draw/RasterYuv420Semiplanar.ooc (hunk 1 above)
Comment (sebastianbaginski) [inline source/draw/RasterYuv420Semiplanar.ooc]: I think it is evaluated only once. It does not seem to be a problem now, but maybe someone would like to change the `size` at some point.
Should I change it anyway ?

Comment (marcusnaslund) [inline source/draw/RasterYuv420Semiplanar.ooc]: This is fine. I think my suggestion actually does *not* work.

Comment (marcusnaslund) [inline source/draw/RasterYuv420Semiplanar.ooc]: https://github.com/ooc-lang/rock/issues/991

Comment (sebastianbaginski) [inline source/draw/RasterYuv420Semiplanar.ooc]: :(

File: test/draw/RasterYuv420SemiplanarTest.ooc (hunk 2 above)
Comment (sebastianbaginski) [inline test/draw/RasterYuv420SemiplanarTest.ooc]: Yeah, this is the well-known "manual verification" or "looks good enough to me" paradigm in software testing.

Comment (marcusnaslund) [inline test/draw/RasterYuv420SemiplanarTest.ooc]: #1853

File: source/draw/RasterYuv420Semiplanar.ooc [hunk 3]
```diff
@@ -92,30 +92,36 @@ RasterYuv420Semiplanar: class extends RasterImage {
...
 		if (target size y isOdd)
 			targetSizeHalf = IntVector2D new(targetSizeHalf x, targetSizeHalf y + 1)
 		for (row in 0 .. targetSizeHalf y) {
-			srcRow := (thisSizeHalf y * row) / targetSizeHalf y
+			srcRow := (thisSizeHalf y * row) / targetSizeHalf y + sourceBox top / 2
 			thisStride := srcRow * this uv stride / 2
 			targetStride := row * target uv stride / 2
 			for (column in 0 .. targetSizeHalf x) {
-				srcColumn := (thisSizeHalf x * column) / targetSizeHalf x
+				srcColumn := (thisSizeHalf x * column) / targetSizeHalf x + sourceBox left / 2
```
Comment (marcusnaslund) [inline source/draw/RasterYuv420Semiplanar.ooc]: Oh, I should ask: You are reading the `sourceBox` properties once per iteration, do you want to speed them up as you did with `thisSize*` and `targetSize*`?

Comment (marcusnaslund) [inline source/draw/RasterYuv420Semiplanar.ooc]: lines 105, 109, 120, 124

Comment (sebastianbaginski) [inline source/draw/RasterYuv420Semiplanar.ooc]: We can do that later, I'm currently still working on this function.

Comment (sebastianbaginski) [inline source/draw/RasterYuv420Semiplanar.ooc]: But thanks for noticing :+1:

Comment (marcusnaslund) [inline source/draw/RasterYuv420Semiplanar.ooc]: It's what they pay me for.

**Pull request merged**